### DISCUSSION TITLE
Not able to establish Multicast using fuota-test-1 application

### DISCUSSION THREAD

**Discussion by shabaz-git** [Lora-net/LoRaMac-node#1099] [Q&A]
Hi,

We are facing an issue in establishing multicasting(FUOTA). 
The logs we are observing are not consistent though we didn't change anything.
Initially, we used to get the MCkeys on the node side but later even we are not able to see it. Once the device got the McKey it's going into a continuous loop and we are getting junk prints over the console. Even DR and freq parameter also zero always though we configured DR =5 and freq = 869525000. 

We have followed the below steps to establish multicast:
1.  fuota-test-1 application flashed on end node.
2. created device on chirpstack with same keys. Node joined in class A. 
3. created a multicast group on chirpstack and added the same device in this group.
4. Finally we created a firmware update job and started.

Please let us know if there are any standard steps to follow to establish multicasting over chirpstack.

Thanks.

**Comment by mluis1** [comment]
Nowadays the ChirpStack server does not have anymore the FUOTA support.
The FUOTA is now managed by an external server connected to the ChirpStack server. Please refer to https://github.com/brocaar/chirpstack-fuota-server

One year ago I have provided the answer at following link https://github.com/Lora-net/LoRaMac-node/issues/827#issuecomment-602721672 where I provided the required steps to get the `fuota-test-1` example to successfully complete.
To run that test I used the ChirpStack version available by then.

It has to be noted that step 3 in your description is not need. The multicast parameters are managed automatically by the FUOTA server.

**Reply by shabaz-git** [reply]
Thanks @mluis1 ,

> One year ago I have provided the answer at following link #827 (comment) where I provided the required steps to get the fuota-test-1 example to successfully complete.

We followed the same and we changed the macros `FRAG_MAX_SIZE` to `255` and `FRAG_MAX_NB` to `370` as our firmware is of size 90Kb. And one more thing the firmware to be selected is the complete one right (not delta firmware)? 

Now we are stuck at MC_SETUP itself, and the node is continuing the scheduled uplinks. and we are getting the error log in chirpstack app-server like this: 
![chirpstackApp-error](https://user-images.githubusercontent.com/70253206/120074568-15acde00-c0bb-11eb-8f97-a0a9ec7ef6c7.PNG)
I think because of this error we are getting stuck at MC_SETUP itself.

**Reply by mluis1** [reply]
It looks like that there is a problem related to Chirpstack server setup.

From the provided error message it looks like that `MulticastGroupID` is set to `nil` which may mean that no multicast session has been created.

As this looks like to be an issue related to the server setup, I would recommend that you post your issue on Chirpstack forum.
https://forum.chirpstack.io/

The `fuota-test-1` example only implements the file transfer. In addition the received file is stored in RAM memory.
In order to have a fully functional FUOTA process one must implement a secure boot loader and maybe store the file image on a non volatile memory.

**Reply by shabaz-git** [reply]
We moved to the latest versions of the chirpstack and fuota server is running in the docker. All the setup and configurations are properly done. Now trying to send the firmware file.

As the B-L072Z-LRWAN1 board is having only 20kb of RAM, there is no way we can send a 90kb firmware file. Right? Is there is any way to do it?
